### PR TITLE
Add per feed user-agent option to config file

### LINKED ISSUE #26: Adding a User agent header to the requests made
Hi there,

first of all, I'd like to thank you for the great project, I was desperate to find a nice RSS scraper service, and have found it! It being written in Rust adds to my excitement. 

I would like to add some features to rsspls that I am needing for my usage, and therefore wanted to ask you if you are open to PRs?

The first one would be the option to add a user-agent header to the requests made as some websites don't allow for an empty one.

I have already added it to my fork and can open a PR if it sounds good to you.

### PR DESCRIPTION
**Pull request by Lcchy**
Resolves #26 

Ps: My IDE auto-formatted the bullet points in the README, please let me know if thats ok or else I can revert.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -215,8 +215,9 @@ async fn process_feed(
         .url
         .parse()
         .wrap_err_with(|| format!("unable to parse {} as a URL", config.url))?;
-    let req = client.get(url.clone());
-    let req = maybe_add_cache_headers(req, cached_headers);
+    let mut req = client.get(url.clone());
+    req = add_headers(req, cached_headers, &channel_config.user_agent);
```
Comment (wezm) [inline src/main.rs:219]: Think I preferred the prior construct where `req` was shadowed and therefore didn't need to be mutable. That would look like this:

```suggestion
    let req = client.get(url.clone());
    let req = add_headers(req, cached_headers, &channel_config.user_agent);
```

Better yet might be:

```suggestion
    let req = add_headers(client.get(url.clone()), cached_headers, &channel_config.user_agent);
```

File: src/main.rs [hunk 2]
```diff
@@ -334,11 +335,16 @@ fn rewrite_urls(doc: &NodeRef, base_url: &url::ParseOptions) -> eyre::Result<()>
...
+fn add_headers(
     mut req: RequestBuilder,
     cached_headers: &Option<HeaderMap>,
+    user_agent: &Option<String>,
 ) -> RequestBuilder {
-    use reqwest::header::{ETAG, IF_MODIFIED_SINCE, IF_NONE_MATCH, LAST_MODIFIED};
+    use reqwest::header::{ETAG, IF_MODIFIED_SINCE, IF_NONE_MATCH, LAST_MODIFIED, USER_AGENT};
+
+    if let Some(ua) = user_agent {
+        req = req.header(USER_AGENT, ua);
```
Comment (wezm) [inline src/main.rs:346]: Maybe chuck a debug log in here like the others.

**Review by wezm — CHANGES_REQUESTED** [review]

**Comment by Lcchy** [comment]
Sounds good! I added your suggestions and updated the PR

File: src/main.rs (hunk 1 above)
Comment (Lcchy) [reply src/main.rs]: Sounds good

**Pull request merged**